packages/core-ethereum/src/commitment.ts

```diff
@@ -57,14 +70,45 @@ async function createCommitmentChain(db: HoprDB, channelId: Hash, setChainCommit
   log('commitment chain initialized')
 }
 
+/**
+ * Simple class encapsulating channel information
+ * used to generate the initial channel commitment.
+ */
+export class ChannelCommitmentInfo {
+  constructor(
+    public readonly channelEntry: ChannelEntry,
+    public readonly chainId: number,
+    public readonly contractAddress: string
+  ) {}
+
+  /**
+   * Generate the initial commitment seed using this channel information and the given
+   * private node key.
+   * @param nodePrivateKey Local node private key.
+   */
+  public createInitialCommitmentSeed(nodePrivateKey: Uint8Array): Uint8Array {
+    const channelSeedInfo = u8aConcat(
+      this.channelEntry.channelEpoch.serialize(),
+      toU8a(this.chainId, 4),
+      this.channelEntry.getId().serialize(),
```

Comment:
`ChannelEntry` includes a bunch of other things, see

https://github.com/hoprnet/hoprnet/blob/d495e91452199df1f2ca431b8bc18fa2ca84a8b6/packages/utils/src/types/channelEntry.ts#L55-L65

Reply:
Probably a good idea to move stuff out of the `ChannelEntry` over to the `ChannelCommitmentInfo` class, rather then referencing the entire `ChannelEntry` from it.

Reply:
I rather meant `ChannelEntry.serialize()` which serializes also `balance` and `commitment` and `closureTime`. These values are subject of change and it is unlikely to recover them

Reply:
I'd still like to keep things clean and have only stuff that is needed for the initial commitment derivation inside the `ChannelCommitmentInfo` class.